components/ai_chat/core/browser/constants.h

```diff
@@ -24,12 +25,21 @@ extern const base::fixed_flat_set<std::string_view, 1>
 inline constexpr uint8_t kMaxPreviewPages = 20;
 extern const char kLeoModelSupportUrl[];
 
-// Llama 3 has a max context length of 8k tokens. Phi 3 Mini and Llama 2 have 4k
-// tokens max. Using Phi 3 Mini as the upper bound, 1 token ≈ 4 characters, so
-// the max context length is ~16k characters. To allow for follow-ups, set max
-// page content length to 60% of this, i.e., 9.6k characters.
-inline constexpr int kCustomModelMaxPageContentLength = 9600;
-inline constexpr int kCustomModelLongConversationCharLimit = 10000;
+/**
+ * Upon registering a custom model, users have the ability to explicitly
+ * provide a context size (in tokens). When present, we'll use this value
+ * to determine the max page content length (in chars). We will assume 4
+ * chars per token. When no context size has been provided, we will
+ * default to a conservative 4k tokens based on common models like Phi 3
+ * Mini and Llama 2 (both have 4k token context limits).
+ */
+inline constexpr size_t kDefaultCharsPerToken = 4;
+inline constexpr float kMaxContentLengthThreshold = 0.6f;
+inline constexpr size_t kCustomModelDefaultContextSize = 4000;
+// If changing, see also
+// browser\resources\settings\brave_leo_assistant_page\model_config_ui.ts
+inline constexpr int32_t kMaxCustomModelContextSize =
+    std::numeric_limits<int32_t>::max();
```

Comment:
I would prefer if we didn't have this constant `kMaxCustomModelContextSize` around, but actually did enforce this check with `base::CheckedNumeric`, because that would be more consistent, and would have less risks of mistakes.